icic_extraction/grid.py

```diff
@@ -48,21 +61,59 @@ def extract_transaction_from_summary_table (summary_table):
     summary_table_data = []
     for transaction  in summary_table:
         clean_transaction = tuple(element.strip() for element in transaction)
+        clean_transaction = tuple(Element.replace("Brokerage Charges","")for Element in clean_transaction)
+        clean_transaction = tuple(Element.replace("Charges","")for Element in clean_transaction)
+        clean_transaction = tuple(Element.replace("and Clearing","")for Element in clean_transaction)
         clean_transaction_lis = list(clean_transaction)
         company_name = clean_transaction_lis.pop(0)
         transaction_record = [x.split() for x in clean_transaction_lis ]
         transaction_record.insert(0,company_name)
         summary_table_data.append(transaction_record)
     return summary_table_data
+
+def clean_trans_summary(sum_update):
+    res = []
+    for i in sum_update:
+        if i[1][1]!= "0":
+            res.append(i[1]+[i[0],i[-1]])
+        if i[2][1] != "0":
+            res.append(i[2]+[i[0],i[-1]])
+    return res
+
+def flatten_transaction_table (meta,clean_trans):
+    return [i+meta for i in clean_trans]
+
 def get_totals_table(contract_cont):
```

Comment:
remove this function

Reply:
get_totals ?

Reply:
yes, get_totals_table 
we are not using it anymore

Reply:
ok